msdk-id-sirius/src/main/java/SiriusIdentificationMethod.java

```diff
@@ -0,0 +1,349 @@
+import org.openscience.cdk.interfaces.IIsotope;
+import org.openscience.cdk.interfaces.IMolecularFormula;
+import org.openscience.cdk.silent.SilentChemObjectBuilder;
+import org.openscience.cdk.tools.manipulator.MolecularFormulaManipulator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p> SiriusIdentificationMethod class. </p>
+ *
+ * This class wraps the Sirius module and transforms its results into MSDK data structures
+ * Transformation of IdentificationResult (Sirius) into IonAnnatation (MSDK)
+ * Containts sub-class for generating FormulaConstraints for Sirius from MolecularFormulaRange object
+ */
+public class SiriusIdentificationMethod implements MSDKMethod<List<IonAnnotation>> {
+
+  /**
+   * Dynamic loading of native libraries
+   */
+  static {
+    try {
+      System.load("glpk_4_60");
+      System.load("glpk_4_60_java");
+    } catch (UnsatisfiedLinkError e) {
+      try {
+        // GLPK requires two libraries
+        String[] libs = {"glpk_4_60", "glpk_4_60_java"};
+        NativeLibraryLoader.loadLibraryFromJar("glpk-4.60", libs);
+      } catch (Exception e1) {
+        throw new RuntimeException(e1);
+      }
+    }
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(SiriusIdentificationMethod.class);
+  private final Sirius sirius;
+  private final List<MsSpectrum> ms1;
+  private final List<MsSpectrum> ms2;
+  private final Double parentMass;
+  private final IonType ion;
+  private final int numberOfCandidates;
+  private final FormulaConstraints constraints;
+  private final Deviation deviation;
+  private boolean cancelled = false;
+  private List<IonAnnotation> result;
+
+  /* Set to be class elements for performance */
+
+  /**
+   * <p> Constructor for SiriusIdentificationMethod class. </p>
+   *
+   * @param ms1 - can be null! MsSpectrum level 1
+   * @param ms2 - MsSpectrum level 2
+   * @param parentMass - Most intensive usually or specified
+   * @param ion - Ionization
+   * @param numberOfCandidates - amount of IdentificationResults to be returned from Sirius
+   * @param constraints - FormulaConstraints provided by the end user. Can be created using ConstraintsGenerator
+   * @param deviation - float value of possible mass deviation
+   */
+  public SiriusIdentificationMethod(@Nullable List<MsSpectrum> ms1, @Nonnull List<MsSpectrum> ms2, Double parentMass,
+      IonType ion, int numberOfCandidates, @Nullable FormulaConstraints constraints, Double deviation) {
+    sirius = new Sirius();
+    this.ms1 = ms1;
+    this.ms2 = ms2;
+    this.parentMass = parentMass;
+    this.ion = ion;
+    this.numberOfCandidates = numberOfCandidates;
+    this.constraints = constraints;
+    this.deviation = new Deviation(deviation);
+  }
+
+  /**
+   * <p> Class ConstraintsGenerator. </p>
+   * This class allows to construct a Sirius object FormulaConstraints using MolecularFormulaRange object
+    */
+  public static class ConstraintsGenerator {
```

Comment:
Is there any advantage in having this class as a static class inside SiriusIdentificationMethod? I would change it to a stand-alone class, it will make SiriusIdentificationMethod shorter and more clear.

Reply:
First I thought about making it as one of the methods of SiriusIdentificationMethod class, but decided that it will be too overloaded.
As it is strongly correlated with Sirius module, I left it here.

But it is not a problem, I can move it.

Reply:
Shorter classes are always better.